File: priv/alphonse/lib/integer.js
import Tuple from './tuple';
import Atom from './atom';

let Integer = {
  __MODULE__: Atom('Integer'),

  is_even: function(n){
    return n % 2 === 0;
  },

  is_odd: function(n){
    return n % 2 !== 0;
  },

  parse: function(bin){
    let result = parseInt(bin);

    if(isNaN(result)){
      return Atom('error');
    }

    let indexOfDot = bin.indexOf(".");

    if(indexOfDot >= 0){
      return Tuple(result, bin.substring(indexOfDot));
    }

    return Tuple(result, "");
  },

  to_char_list: function(number, base = 10){
    return number.toString(base).split('');
  },

  to_string: function(number, base = 10){
    return number.toString(base);
  }
};

export default Integer;

File: lib/std_lib/integer.ex
defmodule ElixirScript.Integer do
  @moduledoc false
  def is_even(number) do
    rem(number, 2) == 0
  end

  def is_odd(number) do
    rem(number, 2) != 0
  end

  def to_char_list(number) do
    to_char_list(number, 10)
  end

  def to_char_list(number, base) do
    number.toString(base).split('')
  end

  def parse(bin) do
    result = Elixir.Core.Functions.get_global().parseInt(bin)

    if Elixir.Core.Functions.get_global().isNaN(result) do
      :error
    else
      case bin.indexOf(".") do
        index_of_dot when index_of_dot < 0 ->
          {result, ""}
        index_of_dot ->
          {result, bin.substring(index_of_dot)}
      end
    end
  end
end
